browser/ui/toolbar/brave_app_menu_model_browsertest.cc

```diff
@@ -486,3 +489,40 @@ IN_PROC_BROWSER_TEST_F(BraveAppMenuBrowserTest, BraveIpfsMenuTest) {
   }
 }
 #endif
+
+void CheckMenuIcons(ui::MenuModel* menu,
+                    int submenu_depth,
+                    std::u16string path = u"") {
+  constexpr int kIconlessCommands[] = {
+      // Header, with no icon
+      IDC_RECENT_TABS_NO_DEVICE_TABS,
+      // Header, with no icon
+      RecentTabsSubMenuModel::kDisabledRecentlyClosedHeaderCommandId};
+  for (size_t i = 0; i < menu->GetItemCount(); ++i) {
```

Comment:
@cdesouza-chromium this will fail if a new menu item is added without an icon - not sure if that's going to make uplifts harder (if it is, I can do something else, like printing a big error message to the console with `DCHECK_IS_ON`). I don't think it's super common to add things to this menu though (and if something is added it's probably going to break the menu anyway, because we customize it a lot).

Reply:
Talked on slack, we're going to see whether it's disruptive